spec/mongoid/traversable_spec.rb

```diff
@@ -251,4 +251,164 @@
       end
     end
   end
+
+  describe "#discriminator_key" do
+
+    context "when the discriminator key is not set on a class" do
+      it "equals _type" do
+        expect(Instrument.discriminator_key).to eq("_type")
+      end
+
+      it "the global discriminator key is _type" do
+        expect(Mongoid.discriminator_key).to eq("_type")
+      end
+
+      it "child discriminator keys equal _type: Piano" do
+        expect(Piano.discriminator_key).to eq("_type")
+      end
+
+      it "child discriminator keys equal _type: Guitar" do
+        expect(Guitar.discriminator_key).to eq("_type")
+      end
+    end
+
+    context "when the discriminator key is changed at the global level" do
+      before do
+        Mongoid.discriminator_key = "hello"
+      end
+
+      after do
+        Mongoid.discriminator_key = "_type"
+      end
+
+      it "the value changes" do
+        expect(Mongoid.discriminator_key).to eq("hello")
+      end
+
+      it "is changed in the parent" do
+        expect(Instrument.discriminator_key).to eq("hello")
+      end
+
+      it "is changed in the child: Piano" do
+        expect(Piano.discriminator_key).to eq("hello")
+      end
+
+      it "is changed in the child: Guitar" do
+        expect(Guitar.discriminator_key).to eq("hello")
+      end
+    end
+
+    context "when the discriminator key is changed in the parent" do
+      before do
+        Instrument.discriminator_key = "hello2"
+      end
+
+      after do
+        Instrument.discriminator_key = nil
+      end
+
+      it "the global discriminator key is _type" do
+        expect(Mongoid.discriminator_key).to eq("_type")
+      end
+
+      it "the value changes" do
+        expect(Instrument.discriminator_key).to eq("hello2")
+      end
+
+      it "changes in the child class: Piano" do
+        expect(Piano.discriminator_key).to eq("hello2")
+      end
+
+      it "changes in the child class: Guitar" do
+        expect(Guitar.discriminator_key).to eq("hello2")
+      end
+
+      context 'when discriminator key is set to nil in parent' do
+        before do
+          Instrument.discriminator_key = nil
+        end
+
+        it "the global discriminator key is _type" do
+          expect(Mongoid.discriminator_key).to eq("_type")
+        end
+
+        it 'uses global setting' do
+          expect(Instrument.discriminator_key).to eq("_type")
+        end
+
+        it "changes in the child class: Piano" do
+          expect(Piano.discriminator_key).to eq("_type")
+        end
+  
+        it "changes in the child class: Guitar" do
+          expect(Guitar.discriminator_key).to eq("_type")
+        end
+      end
+
+      context "when resetting the discriminator key after nil" do
+        before do
+          Instrument.discriminator_key = "hello4"
```

Comment:
I think you need to put a line before this one that sets the discriminator key to nil? Correct me if I'm wrong here.

Reply:
Yeah, that's a good point... The `after` in the previous test sets it to `nil` so that's what I didn't originally put it there... But you're right, I should.